server/controllers/serverSideRender/serverSideRender.js

```diff
@@ -3,22 +3,33 @@ import { renderToString } from 'react-dom/server'
 import { StaticRouter } from 'react-router'
 import { renderStatic } from 'glamor/server'
 import serializeJS from 'serialize-javascript'
+import { Provider } from 'react-redux'
+
 import App from '../../../shared/components/App'
+import configureStore from '../../../shared/redux/configureStore'
 
 export default (req, res) => {
   const context = {}
+  const initialState = {
+    mapReducer: {}
+  }
+
+  const store = configureStore(initialState)
 
   const { html, css, ids = [] } = renderStatic(() => renderToString(
     <StaticRouter location={req.url} context={context}>
-      <App />
+      <Provider store={store}>
+        <App />
+      </Provider>
     </StaticRouter>
   ))
 
   const templateData = {
     title: 'Trekbase',
     initialHtml: html,
     initialCSS: css,
-    initialIds: serializeJS(ids)
+    initialIds: serializeJS(ids),
+    initialStore: serializeJS(store.getState())
```

Comment:
Can we change the `serializeJS` to be `serializeJS(store.getState(), { isJSON: true })` ?
Since our state in redux is serialisable and does not contain any function expressions the `isJSON` flag will give us a 3x performance boost - premature micro optimisation  is generally a bad thing but this doesn't require much effort!